JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -220,8 +220,14 @@ module Microsoft.ApplicationInsights {
         /**
          * helper method to set userId and sessionId cookie
          */
-        public static setCookie(name, value) {
-            Util.document.cookie = name + "=" + value + ";path=/";
+        public static setCookie(name, value, domain?) {
+            var domainAttrib = "";
+
+            if (domain) {
+                domainAttrib = ";domain=" + domain;
```

Comment:
@SergeyKanzhelev with this PR, the customer can specify foo.com in the setting and it will apply to both site1.foo.com and site2.foo.com. This is the use case - allowing cookie to work across subdomains.
Allowing cookie to work across two arbitrary domains without common top level parent (like in your example abc.foo.com and cde.foo.ru) is not the use case for this fix and I don't think it will be technically feasible if it was a use case.
Configuration of different environment be done by adding to the snippet (I image it will be a mix of server and client code) something like this:
`cookieDomain = @Config.CookieDomain;`
or you can have if..else logic here too